pkg/cmd/projectCreate.go

```diff
@@ -0,0 +1,467 @@
+		Modules     []MavenCoordinates
+	}
+
+	// Fill-in all the parameters the template needs.
+	pomTemplateParameters := OBRPomParameters{
+		Parent:      MavenCoordinates{GroupId: packageName, ArtifactId: packageName, Name: packageName},
+		Coordinates: MavenCoordinates{GroupId: packageName, ArtifactId: packageName + ".obr", Name: packageName + ".obr"},
+		Modules:     make([]MavenCoordinates, len(featureNames))}
+	// Populate the list of modules.
+	for index, featureName := range featureNames {
+		pomTemplateParameters.Modules[index] = MavenCoordinates{
+			GroupId: packageName, ArtifactId: packageName + "." + featureName, Name: packageName + "." + featureName}
+	}
+
+	targetFile := GeneratedFile{
+		fileType:                 "pom",
+		targetFilePath:           targetOBRFolderPath + "/pom.xml",
+		embeddedTemplateFilePath: "templates/projectCreate/parent-project/obr-project/pom.xml",
+		templateParameters:       pomTemplateParameters}
+
+	err := createFile(fileSystem, targetFile, forceOverwrite)
+	return err
+}
+
+//---------------------------------------------------------------------------------------------------
+// File generation functions
+//---------------------------------------------------------------------------------------------------
+
+type GeneratedFile struct {
+	fileType                 string
+	targetFilePath           string
+	embeddedTemplateFilePath string
+	templateParameters       interface{}
+}
+
+// checkAllowedToWrite - Checks to see if we are allowed to write a file.
+// The file may exist already. If it does, then we won't be able to over-write it unless
+// the forceOverWrite flag is true.
+func checkAllowedToWrite(fileSystem utils.FileSystem, targetFilePath string, forceOverwrite bool) error {
+	isAlreadyExists, err := fileSystem.Exists(targetFilePath)
+	if err == nil {
+		if isAlreadyExists && (!forceOverwrite) {
+			log.Printf("File %s exists, and we cannot over-write it as the --force flag is not set.", targetFilePath)
+			err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_CANNOT_OVERWRITE_FILE, targetFilePath)
+		}
+	}
+	return err
+}
+
+// createFile creates a file on the file system.
+// If forceOverwrite is false, and there is already a file there, then an error will occur.
+func createFile(
+	fileSystem utils.FileSystem,
+	generatedFile GeneratedFile,
+	forceOverwrite bool) error {
+
+	log.Printf("Creating file of type %s at %s\n", generatedFile.fileType, generatedFile.targetFilePath)
+
+	err := checkAllowedToWrite(fileSystem, generatedFile.targetFilePath, forceOverwrite)
+	if err == nil {
+		var template *template.Template
+		template, err = loadEmbeddedTemplate(generatedFile.embeddedTemplateFilePath)
+		if err == nil {
+			var fileContents string
+			fileContents, err = substituteParametersIntoTemplate(template, generatedFile.templateParameters)
+			if err == nil {
+				// Write it out to the target file.
+				err = fileSystem.WriteTextFile(generatedFile.targetFilePath, fileContents)
+			}
+		}
```

Comment:
fileContents must be defined, so that forces it to be in the nest, unless I move the fileContents above, and set it nil. If it was more nested, then I would break up.